### PR TITLE
Add methods for read and write of PyRegion

### PR DESCRIPTION
**Pull request by chetan51**
Fixes https://github.com/numenta/nupic/issues/2678.

### CODE REVIEW

**Comment by chetan51** [comment]
@scottpurdy

File: src/nupic/regions/PyRegion.py [hunk 1]
```diff
@@ -273,6 +273,25 @@ def setParameterArray(self, name, index, array):
...
     """
 
+
+  def write(self, proto):
+    """This method is called during network serialization with a proto object
+    so that this region can write its data to it.
+
+    proto: PyRegionProto capnproto object
+    """
+    pass
```
Comment (scottpurdy) [inline src/nupic/regions/PyRegion.py]: Should this raise NotImplemented?

**Comment by scottpurdy** [comment]
Should we just delete the TestNode in `nupic` since we have one in the bindings?

File: src/nupic/regions/PyRegion.py (hunk 1 above)
Comment (chetan51) [inline src/nupic/regions/PyRegion.py]: I don't think so, because there can be subclasses of this that don't need serialization.

**Comment by scottpurdy** [comment]
:+1:

**Comment by chetan51** [comment]
The `TestNode` in `nupic` is used in `network_test.py`.

**Pull request merged**